homeassistant/components/tplink/binary_sensor.py

```diff
@@ -0,0 +1,88 @@
+"""Support for TPLink binary sensors."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final
+
+from kasa import Feature
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import CoordinatedTPLinkFeatureEntity, TPLinkFeatureEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkBinarySensorEntityDescription(
+    BinarySensorEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+BINARYSENSOR_DESCRIPTIONS: Final = (
+    TPLinkBinarySensorEntityDescription(
+        key="overheated",
+        device_class=BinarySensorDeviceClass.HEAT,
```

Comment:
Shouldn't this be the `problem` device class instead?

Reply:
Yeah, I will change that, I just picked it based on the docs (`On means hot, Off means normal.`).

From a brief grep, looks like only `fibaro` is using this for fire sensors / overheating and it's otherwise used for heating related entities.

Reply:
Done.